### ISSUE TITLE
custom keybindings fail with urwid 2.0.1 and python3.8

### ISSUE THREAD

**Issue by flaper87** [vit-project/vit#217]
I've installed vit from the `2.x` branch and my keybindings seem to be failing because of, what I believe is, an incompatibility with urwid 2.0.1

```
$ vit --version
2.0.0
```

```
Traceback (most recent call last):
  File "/usr/bin/vit", line 11, in <module>
    load_entry_point('vit==2.0.0', 'console_scripts', 'vit')()
  File "/usr/lib/python3.8/site-packages/vit-2.0.0-py3.8.egg/vit/command_line.py", line 5, in main
    Application(options, filters)
  File "/usr/lib/python3.8/site-packages/vit-2.0.0-py3.8.egg/vit/application.py", line 74, in __init__
    self.loop.run()
  File "/usr/lib/python3.8/site-packages/urwid/main_loop.py", line 286, in run
    self._run()
  File "/usr/lib/python3.8/site-packages/urwid/main_loop.py", line 384, in _run
    self.event_loop.run()
  File "/usr/lib/python3.8/site-packages/urwid/main_loop.py", line 788, in run
    self._loop()
  File "/usr/lib/python3.8/site-packages/urwid/main_loop.py", line 825, in _loop
    self._watch_files[fd]()
  File "/usr/lib/python3.8/site-packages/urwid/raw_display.py", line 403, in <lambda>
    wrapper = lambda: self.parse_input(
  File "/usr/lib/python3.8/site-packages/urwid/raw_display.py", line 502, in parse_input
    callback(processed, processed_codes)
  File "/usr/lib/python3.8/site-packages/urwid/main_loop.py", line 411, in _update
    self.process_input(keys)
  File "/usr/lib/python3.8/site-packages/urwid/main_loop.py", line 517, in process_input
    something_handled |= bool(self.unhandled_input(k))
  File "/usr/lib/python3.8/site-packages/urwid/main_loop.py", line 563, in unhandled_input
    return self._unhandled_input(input)
  File "/usr/lib/python3.8/site-packages/vit-2.0.0-py3.8.egg/vit/application.py", line 341, in key_pressed
    self.execute_macro(keys)
  File "/usr/lib/python3.8/site-packages/vit-2.0.0-py3.8.egg/vit/application.py", line 226, in execute_macro
    keypresses = self.prepare_keybinding_keypresses(keybinding['keys'])
  File "/usr/lib/python3.8/site-packages/vit-2.0.0-py3.8.egg/vit/application.py", line 236, in prepare_keybinding_keypresses
    return reduce(reducer, keypresses, [])
  File "/usr/lib/python3.8/site-packages/vit-2.0.0-py3.8.egg/vit/application.py", line 232, in reducer
    accum += list(key())
TypeError: 'bool' object is not iterable
```

**Comment by thehunmonkgroup** [comment]
That's a Python error, most likely due to a mis-configuration of your keybindings.

**Comment by flaper87** [comment]
> That's a Python error, most likely due to a mis-configuration of your keybindings.

This is the keybinding I'm trying to use (which used to work before I updated to latest master)

`o = :!wr /home/flaper87/.bin/task-pr.sh {TASK_UUID}<Enter>`

What am I missing?

**Comment by nfraprado** [comment]
If that started happening after updating to python 3.8, it may be caused by the same reason that [this tasklib bug](https://github.com/robgolding/tasklib/issues/67).
As shown there, it was fixed by [this pull request](https://github.com/robgolding/tasklib/pull/68), which was included in version 1.3.0 of tasklib. So if you haven't already, please update to tasklib 1.3.0 and see if that fixes your problem.

**Comment by flaper87** [comment]
> If that started happening after updating to python 3.8, it may be caused by the same reason that [this tasklib bug](https://github.com/robgolding/tasklib/issues/67).
> As shown there, it was fixed by [this pull request](https://github.com/robgolding/tasklib/pull/68), which was included in version 1.3.0 of tasklib. So if you haven't already, please update to tasklib 1.3.0 and see if that fixes your problem.

Thanks! I think it's indeed related to the upgrade to 3.8. I'll update `tasklib` :)
